cmd/generic-handlers.go

```diff
@@ -550,6 +550,9 @@ func addCustomHeaders(h http.Handler) http.Handler {
 		// part of the log entry, Error response XML and auditing.
 		// Set custom headers such as x-amz-request-id for each request.
 		w.Header().Set(xhttp.AmzRequestID, mustGetRequestID(UTCNow()))
+		if globalLocalNodeName != "" {
+			w.Header().Set(xhttp.AmzRequestNodeID, globalLocalNodeName)
```

Comment:
Exposing internal hostnames seems problematic. Could we set this to server-pool+server index?

Reply:
@klauspost we could do that